src/stories/Button.tsx

```diff
@@ -0,0 +1,50 @@
+import React, { useMemo } from "react";
+
+const getVariantClasses = (variant: string) => {
+  switch (variant) {
+    case "filled": {
+      return "text-white bg-purple-800 border-purple-800 border-2";
+    }
+    case "outlined": {
+      return "text-purple-800 bg-transparent border-slate-400 border-2";
+    }
+    case "text": {
+      return "text-purple-800 bg-transparent border-transparent border-2";
+    }
+    case "tonal": {
+      return "text-slate-800 bg-purple-200 border-transparent border-2";
+    }
+    case "elevated": {
+      return "text-purple-800 bg-slate-100 border-grey-200 border-2 shadow-md";
+    }
+    default: {
+      return "text-white bg-purple-800 border-purple-800 border-2";
+    }
+  }
+};
+
+const BASE_BUTTON_CLASSES = "cursor-pointer rounded-full px-4 py-2";
+
+export const Button = ({
+  variant = "filled",
+  label = "Button",
+  // showIcon = false,
+  // icon = null,
+  ...props
+}) => {
```

Comment:
Should there be a type declaration here?

```tsx
type ButtonProps = {
  /**
  What styles should be applied to the element?
  */
  variant?:
    | 'filled'
    | 'outlined'
    | 'text'
    | 'tonal'
    | 'elevated'

  /**
  What the button label should be.
  */
  label?: string;

  /**
  What custom classes should be applied to the element?
  */
  className?: string;

  /**
  When present, specifies that the button should be disabled.
  */
  disabled?: boolean;

  /**
  What the inner content of the button is.
  */
  children?: React.ReactNode;

  /**
  Optional click handler.
  */
  onClick?: () => void;
};

...

export const Button = ({
  variant = "filled",
  label = "Button",
  // showIcon = false,
  // icon = null,
  ...props
}: ButtonProps) => {

  ...

}
```

Reply:
You're right, thanks for catching that -- I'll make the fix!